com.ibm.streamsx.topology/opt/python/include/splpy_general.h

```diff
@@ -187,7 +187,36 @@ class SplpyGeneral {
       return ret;
     }
 
-    
+    /**
+     * Utility method to write a python exception to the application trace.
+     * The type and value are written, but not the traceback.
+     * If no python exception occurred, this does nothing.
+     * The caller must hold the GILState.
+     */
+    static void tracePythonError() {
```

Comment:
We already trace Python errors, so why is a new function needed?

Reply:
Python errors are written to standard error, but not the application trace.  This will not preserve important information for diagnosing problems, such as the time the error occurred and the operator it occurred in.